userspace/plugin/plugin_types.h

```diff
@@ -293,6 +293,37 @@ typedef enum ss_plugin_log_severity
 	SS_PLUGIN_LOG_SEV_TRACE = 8,
 } ss_plugin_log_severity;
 
+// Types supported by the by the metric values
+typedef enum ss_plugin_metric_value_type
+{
+	SS_PLUGIN_METRIC_VALUE_TYPE_U32 = 1,
+	SS_PLUGIN_METRIC_VALUE_TYPE_S32 = 2,
+	SS_PLUGIN_METRIC_VALUE_TYPE_U64 = 3,
+	SS_PLUGIN_METRIC_VALUE_TYPE_S64 = 4,
+	SS_PLUGIN_METRIC_VALUE_TYPE_D = 5,
+	SS_PLUGIN_METRIC_VALUE_TYPE_F = 6,
+	SS_PLUGIN_METRIC_VALUE_TYPE_I = 7,
+} ss_plugin_metric_value_type;
+
+// Data representation of metric values
+typedef union ss_plugin_metric_value 
+{
+	uint32_t u32;
+	int32_t s32;
+	uint64_t u64;
+	int64_t s64;
+	double d;
+	float f;
+	int i;
+} ss_plugin_metric_value;
+
+// Metric types
+typedef enum ss_plugin_metric_type
+{
+	SS_PLUGIN_METRIC_TYPE_COUNTER = 1,
```

Comment:
For instance here we introduce different types, which will be more difficult to maintain and it could confuse other contributors. Could we consider just using the existing definitions in `metrics_v2.h` (see previous comment)?

Reply:
As I said in the previous comment, I think we need to redefine these types.
However in this particular case, I agree that different names can be confusing and we should at least make these types have the same names as the ones in `metrics_v2`.

Reply:
See https://github.com/falcosecurity/libs/pull/1828#issuecomment-2103224934